src/openms_gui/source/VISUAL/APPLICATIONS/TOPPViewBase.cpp

```diff
@@ -1581,21 +1581,6 @@ namespace OpenMS
     if (w)
     {
       intensity_button_group_->button(index)->setChecked(true);
-      Spectrum2DWidget* w2d = dynamic_cast<Spectrum2DWidget*>(w);
```

Comment:
nope, i've moved this functionality to Spectrum2DCanvas::intensityModeChange_() and also fixed the bug there that only the currently active layer changes its gradient in log-mode vs non-log mode.